contracts/script/README.md

```diff
@@ -0,0 +1,142 @@
+# Deploying Upgradeable Smart Contracts
+
+Upgradeable Smart contracts are deployed with Openzeppelin Defender to enable a deployment strategy that is more secure
+and also uses a multi-sig Safe wallet. When deploying using openzeppelin the `defender` profile in the `foundry.toml`
+file is used.
+
+Note: One must set the environment variable `FOUNDRY_PROFILE` accordingly to use the appropriate profile in the
+`foundry.toml` settings. E.g.
+
+```bash
+export FOUNDRY_PROFILE=defender
+```
+
+The profile, `profile.default` is used by default. Setting the `FOUNDRY_PROFILE` variable overrides the `foundry.toml`
+settings.
+
+## Prerequisites
+
+1. Create a multisig wallet using [Safe](https://app.safe.global/welcome/accounts) on the network you'd like to deploy
+   to.
+2. In OpenZeppelin Defender, create an Approval Process that requires the multisig wallet you created above.
+   `Manage > Approval Processes`
+3. In OpenZeppelin Defender, create a deployment environment and use the approval process created in Step 2. Be sure to
+   copy the Defender secret and key, that is shown at the end of this step, into the .env file.
+
+## Deployments
+
+### Deploying the Fee Contract
+
+Steps:
+
+1. Run the Deployment Command This command requires you to go to OpenZeppelin Defender's UI to see the transaction.
+   Click that transaction which opens up the Safe UI where your signers for that Safe multi-sig wallet can confirm the
+   transaction. The two transactions to be confirmed are: (i) deployment of implementation contract (ii) deployment of
+   proxy contract
+
+```bash
+export FOUNDRY_PROFILE=defender && \
```

Comment:
Is this required for `forge build`?

If not, 

```
env FOUNDRY_PROFILE=defender forge script contracts/script/FeeContractWithDefender.s.sol:FeeContractDefenderDeployScript --ffi --rpc-url https://ethereum-sepolia.publicnode.com
```
Would only set the env var for the command, so there's no need to restore it at the end.

Reply:
Both profiles (default and defender) work for forge build but generating the bindings fail for the defender profile. The defender profile sets build info to true and the existence of the build info in the generated json for the contracts causes the binding generation to fail. (so this is what inspired creating a separate foundry profile for running defender scripts)

Reply:
Is there an open issue about this on the foundry github repo? If not we should open one.

Reply:
So the issue is that `build_info = true` creates an extra file in the `out` folder that forge bind tries to generate bindings from.

Reply:
Upstream issue:
- https://github.com/foundry-rs/foundry/issues/7443

Reply:
@alysiahuggins https://github.com/foundry-rs/foundry/pull/7444 to fix this issue has been merged. We can temporarily switch to using nightly foundry until this fix lands in the monthly release in order to avoid having to have two foundry profiles.

In about 15 hours [foundry.nix](https://github.com/shazow/foundry.nix) should be updated and at that point we can also use a foundry version with this fix in flake.nix by changing

https://github.com/EspressoSystems/espresso-sequencer/blob/29edcb1cf2ebfa4c54a1773cec07844d2d1fb5e6/flake.nix#L26-L27

to 

```
  inputs.foundry.url = "github:shazow/foundry.nix/main";
```

The CI is already using nightly so shouldn't need to update any of the workflows.